src/threading.cpp

```diff
@@ -831,8 +833,119 @@ gb_internal void futex_wait(Futex *f, Footex val) {
 		WaitOnAddress(f, (void *)&val, sizeof(val), INFINITE);
 	} while (f->load() == val);
 }
+
+#elif defined(GB_SYSTEM_HAIKU)
+
+// Futex implementation taken from https://tavianator.com/2023/futex.html
+
+#include <pthread.h>
+#include <atomic>
+ 
+struct Futex_Wait_Node {
+	pthread_t thread;
+	Futex *futex;
+	Futex_Wait_Node *prev, *next;	
+};
+ 
+struct Futex_Wait_Queue {
+	std::atomic_flag spinlock;
+	Futex_Wait_Node list;
+ 
+	void lock() {
+		while (spinlock.test_and_set(std::memory_order_acquire)) {
+			; // spin...
```

Comment:
At least put a cpu_relax in there of some sort.

Reply:
is there an alternative for this in the compiler codebase?

Reply:
`intrinsics.cpu_relax()`